### PR TITLE
Make `test_generate_with_static_cache` even less flaky

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

#34464 make some generate tests less flaky. However, with 

> tests/models/paligemma/test_modeling_paligemma.py::PaliGemmaForConditionalGenerationModelTest::test_generate_with_static_cache

it fails 0.6% of the time.

This PR takes the same idea in #34558 to make `test_generate_with_static_cache` even much less flaky. Running 2000 times and all pass.

It also make this process easier to applied to other tests in the future whenever necessary

### CODE REVIEW

**Comment by ydshieh** [comment]
also fixed

> tests/models/persimmon/test_modeling_persimmon.py::PersimmonModelTest::test_generate_with_static_cache

File: tests/models/seamless_m4t_v2/test_modeling_seamless_m4t_v2.py [hunk 1]
```diff
@@ -835,7 +835,12 @@ def test_generation_languages(self):
     def test_speech_generation(self):
```
Comment (ydshieh) [inline tests/models/seamless_m4t_v2/test_modeling_seamless_m4t_v2.py:835]: > tests/models/seamless_m4t_v2test_modeling_seamless_m4t_v2.py::SeamlessM4Tv2GenerationTest::test_speech_generation

previously failed 50% of the time. Now running 100 times and all pass.
(but probably still a bit flaky)

File: tests/models/musicgen_melody/test_modeling_musicgen_melody.py
Comment (ydshieh) [inline tests/models/musicgen_melody/test_modeling_musicgen_melody.py:1]: Run 500 times

> tests/models/musicgen_melody/test_modeling_musicgen_melody.py::MusicgenMelodyDecoderTest::test_eager_matches_sdpa_inference_1_bfloat16

2 failed (before) v.s. 0 failed with PR

> tests/models/musicgen_melody/test_modeling_musicgen_melody.py::MusicgenMelodyTest::test_eager_matches_sdpa_inference_1_bfloat16

3 failed (before) v.s. 0 failed with PR

**Comment by ydshieh** [comment]
> tests/models/persimmon/test_modeling_persimmon.py::PersimmonModelTest::test_generate_with_static_cache

200 runs: 8 failed (before) v.s 0 failed (PR)

**Review by ArthurZucker — APPROVED** [review]
Cool, I like that we have some utils that can be re-used!

**Pull request merged**